webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -498,6 +488,24 @@ def _define_style_ensemble(index: int, x_axis: pd.DataFrame) -> Dict:
     return style
 
 
+def _update_corr_index_dict(
+    response_x_axis: Union[pd.Index, list], observation_x_axis: pd.Index
+) -> int:
+    x_axis_default_observation = _get_first_observation_x(observation_x_axis)
+    if isinstance(response_x_axis, pd.Index):
+        updated_index = response_x_axis.get_loc(x_axis_default_observation)
+    elif isinstance(response_x_axis, list):
+        updated_index = response_x_axis.index(x_axis_default_observation)
+
+    return updated_index
+
+
+def _layout_figure() -> dict:
+    layout = assets.ERTSTYLE["figure"]["layout"].copy()
+    layout.update(dict(showlegend=False))
+    return layout
+
+
 def _get_first_observation_x(obs_data: pd.DataFrame) -> Union[int, str]:
```

Comment:
again not part of the PR, but part of the test sweep and can be updated. It be funny if all this is tests created prior to start refactoring.

deal with exeption to the core function first. then you can emplement the the function afterwards with ease. 
And this function can also use dict (my prefrence over multiple `if-elif-else` statments) to choose what to return.
no need for `if-elif-else` if we are returning after each condition.
suggestion:
```python
data_point = obs_data["x_axis"][0]
caster = {str: int, pd._libs.tslibs.timestamps.Timestamp: str}
if type(data_point) not in caster.keys():
    raise ValueError
return caster.get(type(data_point))(data_point)
```

Reply:
Nice suggestion! I copied pasted shamelessly, only with a minor change to make mypy happy :-) Thanks!